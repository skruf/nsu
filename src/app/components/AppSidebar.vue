<i18n>
{
  "en": {
    "events": "Events",
    "all": "All",
    "upcoming": "Upcoming",
    "history": "History",
    "categories": "Categories",
    "clubs": "Clubs",
    "classes": "Classes",
    "ranges": "Ranges",
    "records": "Records",
    "settings": "Settings"
  },
  "no": {
    "events": "Stevner",
    "all": "Alle",
    "upcoming": "Kommende",
    "history": "Avsluttet",
    "categories": "Kategorier",
    "clubs": "Klubber",
    "classes": "Klasser",
    "ranges": "Skyttebaner",
    "records": "Rekorder",
    "settings": "Innstillinger"
  }
}
</i18n>

<style lang="stylus">
#app
  padding-left 180px
  &.sidebar-open
    padding-left 64px
    i.menu-icon
      margin-right 0
    .app_aside
      width 64px !important
    .logo
      max-width 40px
    .el-menu
      padding-left 0
      text-align center
    .el-menu-item,
    .el-submenu__title
      margin-bottom 10px
    .menu-img
      width 100%
      margin-right 0

.el-submenu .el-menu-item
  padding-left 56px !important
  font-size 13px

.menu-img
  width 23px
  margin-right 8px
  filter invert(.5) sepia(1) saturate(200) hue-rotate(380deg)

.logo
  max-width 84px
  padding 14px 0
  margin-top 25px
  margin-bottom 10px

.el-button.collapse-button
  position fixed
  bottom 0
  left 0
  right 0
  border none
  background-color transparent
  box-shadow none
  &, &:active
    top auto
  i.menu-icon
    margin-right 0

.el-menu.el-menu--popup
  background-color var(--bg-color)

.app_aside
  width 180px !important
  display flex
  flex-direction column
  position absolute
  left 0
  top 0
  bottom 0
  background-color var(--bg-alt-color)
  box-shadow var(--box-shadow)

</style>

<template>
  <el-aside
    width="auto"
    class="app_aside"
  >
    <img
      class="logo mx-auto block"
      src="../../../static/img/nsu-logo.png"
    >

    <el-menu
      default-active="1-1"
      :collapse="isOpen"
    >
      <el-submenu index="1">
        <template slot="title">
          <img
            class="menu-img"
            src="../../../static/img/event-icon.svg"
          >
          <span>{{ $t("events") }}</span>
        </template>
        <el-menu-item
          index="1-1"
          @click="navTo('/events')"
        >
          {{ $t("all") }}
        </el-menu-item>
        <el-menu-item
          index="1-2"
          @click="navTo('/events?filter=upcoming')"
        >
          {{ $t("upcoming") }}
        </el-menu-item>
        <el-menu-item
          index="1-3"
          @click="navTo('/events?filter=history')"
        >
          {{ $t("history") }}
        </el-menu-item>
        <el-menu-item
          index="1-4"
          @click="navTo('/events/categories')"
        >
          {{ $t("categories") }}
        </el-menu-item>
      </el-submenu>

      <el-menu-item
        index="2"
        @click="navTo('/clubs')"
      >
        <img
          class="menu-img"
          src="../../../static/img/club-icon.svg"
        >
        <span>{{ $t("clubs") }}</span>
      </el-menu-item>

      <el-menu-item
        index="3"
        @click="navTo('/classes')"
      >
        <img
          class="menu-img"
          src="../../../static/img/class-icon.svg"
        >
        <span>{{ $t("classes") }}</span>
      </el-menu-item>

      <el-menu-item
        index="4"
        @click="navTo('/ranges')"
      >
        <img
          class="menu-img"
          src="../../../static/img/range-icon.svg"
        >
        <span>{{ $t("ranges") }}</span>
      </el-menu-item>

      <el-menu-item
        index="5"
        disabled
        @click="navTo('/records')"
      >
        <img
          class="menu-img"
          src="../../../static/img/record-icon.svg"
        >
        <span>{{ $t("records") }}</span>
      </el-menu-item>

      <el-menu-item
        index="6"
        @click="navTo('/settings')"
      >
        <img
          class="menu-img"
          src="../../../static/img/settings-icon.svg"
        >
        <span>{{ $t("settings") }}</span>
      </el-menu-item>
    </el-menu>

    <el-button
      class="button--text collapse-button"
      @click="toggle"
    >
      <i
        v-if="isOpen"
        class="menu-icon el-icon-arrow-right"
      />
      <i
        v-else
        class="menu-icon el-icon-arrow-left"
      />
    </el-button>
  </el-aside>
</template>

<script>
export default {
  name: "AppSidebar",

  props: {
    open: { type: Boolean, default: true }
  },

  data: function() {
    return {
      isOpen: this.open
    }
  },

  watch: {
    open(open) {
      this.isOpen = open
      this.$emit("update:open", open)
    }
  },

  methods: {
    toggle() {
      this.isOpen = !this.isOpen
      this.$emit("update:open", this.isOpen)
    },

    navTo(url) {
      this.$router.push(url)
    }
  }
}
</script>
