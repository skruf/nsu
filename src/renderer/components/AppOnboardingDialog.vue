<style lang="stylus">
.app-onboarding-dialog
  max-width 420px
  margin-top 15vh !important

  .el-dialog__header
    padding 2px

  .el-carousel__indicators
    bottom 5px

  .el-carousel__indicator
    .el-carousel__button
      background-color var(--primary-alt-color)
    &.is-active
      .el-carousel__button
        background-color var(--primary-color)

.onboarding-slide
  display flex
  flex-direction column
  padding 20px
  height 100%

.onboarding-slide-image
  height 260px

.onboarding-slide-text
  flex 1
  display flex
  align-items center
  justify-content center
  flex-direction column
  text-align center
  // padding-bottom 10px

.onboarding-slide-text-title
  margin-bottom 20px
  line-height 1.25

.onboarding-slide-text-paragraph
  padding 0 10px
  line-height 1.75

</style>

<template>
  <el-dialog
    custom-class="app-onboarding-dialog"
    :append-to-body="true"
    :visible.sync="visible"
    @close="close"
  >
    <el-carousel
      height="560px"
      trigger="click"
      :autoplay="false"
    >
      <el-carousel-item
        v-for="(slide, index) in slides"
        :key="index"
      >
        <div class="onboarding-slide">
          <img
            class="onboarding-slide-image"
            :src="slide.imageUrl"
          >
          <div class="onboarding-slide-text">
            <h2 class="h2 onboarding-slide-text-title">
              {{ slide.title }}
            </h2>
            <p class="paragraph onboarding-slide-text-paragraph">
              {{ slide.text }}
            </p>
          </div>
        </div>
      </el-carousel-item>
    </el-carousel>

    <template slot="footer">
      <el-button
        class="block mx-auto"
        type="primary"
        @click="close"
      >
        Ok, let's go!
      </el-button>
    </template>
  </el-dialog>
</template>

<script>
export default {
  name: "AppOnboardingDialog",

  props: {
    shown: { type: Boolean, default: false }
  },

  data: function() {
    return {
      visible: this.shown,
      slides: [{
        title: "Welcome to NSUs Event Management app!",
        text: "This application is the result of a join effort between NSU and the creators, where the goal is to effectivize workflows for managing shooting competitions inside an organization.",
        imageUrl: "static/img/onboarding-slide-1.svg"
      }, {
        title: "Manage clubs & their members",
        text: "Easily keep track of and manage clubs and their member's participation in events. Later additions of this app will include statistics over the most active participants and their results.",
        imageUrl: "static/img/onboarding-slide-2.svg"
      }, {
        title: "Stay up to date on weapon class rules",
        text: "Rules are important to any competitions, but can sometimes be hard to manage. We have implemented functions for easily keeping track of rules for each weapon class.",
        imageUrl: "static/img/onboarding-slide-3.svg"
      }, {
        title: "Great, now how do I get started?",
        text: "Before you get started with event creation, we recommend that you populate the required clubs and shooting ranges tables first. These items are required in order to create new events.",
        imageUrl: "static/img/onboarding-slide-4.svg"
      }]
    }
  },

  watch: {
    shown(shown) {
      this.visible = shown
      this.$emit("update:shown", shown)
    }
  },

  methods: {
    close() {
      this.visible = false
      this.$emit("update:shown", false)
      this.$emit("close")
    }
  }
}
</script>
