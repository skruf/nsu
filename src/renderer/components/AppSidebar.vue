<style lang="stylus">
#app
  padding-left 180px
  &.sidebar-open
    padding-left 64px
    i.menu-icon
      margin-right 0
    .app_aside
      width 64px !important
    .logo
      max-width 40px
    .el-menu
      padding-left 0
      text-align center

.logo
  max-width 84px
  padding 14px 0
  margin-top 25px
  margin-bottom 10px

.el-button.collapse-button
  position fixed
  bottom 0
  left 0
  right 0
  border none
  background-color transparent
  box-shadow none
  &, &:active
    top auto
  i.menu-icon
    margin-right 0

.el-menu.el-menu--popup
  background-color var(--bg-color)

.app_aside
  width 180px !important
  display flex
  flex-direction column
  position absolute
  left 0
  top 0
  bottom 0
  background-color var(--bg-alt-color)

</style>

<template>
  <el-aside
    width="auto"
    class="app_aside"
  >
    <img
      class="logo mx-auto block"
      src="https://norsksvartkruttunion.no/test/wp-content/uploads/2014/03/nsu-logo-full.png"
    >

    <el-menu
      default-active="1-1"
      :collapse="isOpen"
    >
      <el-submenu index="1">
        <template slot="title">
          <i class="menu-icon ion-logo-ionic" />
          <span>Events</span>
        </template>
        <el-menu-item
          index="1-1"
          @click="navTo('/events')"
        >
          All
        </el-menu-item>
        <el-menu-item
          index="1-2"
          @click="navTo('/events?filter=upcoming')"
        >
          Upcoming
        </el-menu-item>
        <el-menu-item
          index="1-3"
          @click="navTo('/events?filter=history')"
        >
          History
        </el-menu-item>
        <el-menu-item
          index="1-4"
          disabled
          @click="navTo('/events/categories')"
        >
          Categories
        </el-menu-item>
      </el-submenu>

      <el-menu-item
        index="2"
        @click="navTo('/classes')"
      >
        <i class="menu-icon ion-ios-nuclear" />
        <span>Classes</span>
      </el-menu-item>

      <el-menu-item
        index="3"
        @click="navTo('/clubs')"
      >
        <i class="menu-icon ion-ios-people" />
        <span>Clubs</span>
      </el-menu-item>

      <el-menu-item
        index="4"
        @click="navTo('/ranges')"
      >
        <i class="menu-icon ion-ios-people" />
        <span>Ranges</span>
      </el-menu-item>

      <el-menu-item
        index="5"
        disabled
        @click="navTo('/records')"
      >
        <i class="menu-icon ion-ios-star" />
        <span>Records</span>
      </el-menu-item>

      <el-menu-item
        index="6"
        @click="navTo('/settings')"
      >
        <i class="menu-icon ion-ios-settings" />
        <span>Settings</span>
      </el-menu-item>
    </el-menu>

    <el-button
      class="button--text collapse-button"
      @click="toggle"
    >
      <i
        v-if="isOpen"
        class="menu-icon ion-ios-expand"
      />
      <i
        v-else
        class="menu-icon ion-ios-contract"
      />
    </el-button>
  </el-aside>
</template>

<script>
export default {
  name: "AppSidebar",

  props: {
    open: { type: Boolean, default: true }
  },

  data: function() {
    return {
      isOpen: this.open
    }
  },

  watch: {
    open(open) {
      this.isOpen = open
      this.$emit("update:open", open)
    }
  },

  methods: {
    toggle() {
      this.isOpen = !this.isOpen
      this.$emit("update:open", this.isOpen)
    },

    navTo(url) {
      this.$router.push(url)
    }
  }
}
</script>
