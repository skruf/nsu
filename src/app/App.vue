<i18n>
{
  "en": {
    "appWarning": "This is a preview of the software and therefore errors might occur."
  },
  "no": {
    "appWarning": "Dette er en forhåndsvisning av programvaren og feil kan derfor oppstå."
  }
}
</i18n>

<style lang="stylus">
#app
  flex 1
  min-height 100vh
  display flex
  // &:before
  //   content ""
  //   height 4px
  //   width 100%
  //   background-image var(--accent-border)
  //   position absolute
  //   top 0
  //   left 0
  //   right 0
  //   z-index 1

.titlebar
  -webkit-user-select none
  -webkit-app-region drag
  width 100%
  height 20px
  position absolute
  left 0
  right 0
  top 0

.app-warning
  z-index 1
  height 25px
  padding 0
  border-radius 0
  margin-bottom -25px
  background-color #ffca66
  color #000000
  .el-alert__title
    font-size 12px
    font-weight 400
    color inherit
  .el-alert__closebtn
    top 6px
    font-size 14px
    color inherit
.el-alert--warning .el-alert__description
  color inherit

// #app
//   padding-top 25px

</style>

<template>
  <div>
    <!-- <el-alert
      class="app-warning"
      type="warning"
      :title="$t('appWarning')"
      :closable="false"
      center
    /> -->

    <el-container
      id="app"
      :class="{
        'sidebar-open': isSidebarOpen
      }"
    >
      <div class="titlebar" />

      <app-sidebar
        :open.sync="isSidebarOpen"
      />

      <!-- <app-onboarding-dialog
        :shown.sync="appOnboardingShowDialog"
      /> -->

      <router-view />
    </el-container>
  </div>
</template>

<script lang="ts">
// import AppOnboardingDialog from "~/components/AppOnboardingDialog"
import AppSidebar from "~/components/AppSidebar"

export default {
  name: "Nsu",

  components: {
    // AppOnboardingDialog,
    AppSidebar
  },

  data: () => ({
    isSidebarOpen: true
    // appOnboardingShowDialog: true
  }),

  watch: {
    isSidebarOpen() {
      localStorage.setItem("isSidebarOpen", this.isSidebarOpen)
    },
    appOnboardingShowDialog() {
      localStorage.setItem("appOnboardingShowDialog", this.appOnboardingShowDialog)
    }
  },

  created() {
    // this.$electron.ipcRenderer.send("APP_STARTED")

    if(localStorage.isSidebarOpen !== undefined) {
      this.isSidebarOpen = localStorage.isSidebarOpen === "true"
    }
    if(localStorage.appOnboardingShowDialog !== undefined) {
      this.appOnboardingShowDialog = localStorage.appOnboardingShowDialog === "true"
    }

    // this.$electron.ipcRenderer.on("SET_UPDATE_STATUS", (e, status) => {
    //   let title = "Update"
    //   let message
    //
    //   switch(status) {
    //     case "UPDATE_CHECKING":
    //       message = "Checking for updates"
    //       break
    //     case "UPDATE_AVAILABLE":
    //       message = "A new update is available!"
    //       break
    //     case "UPDATE_NOT_AVAILABLE":
    //       message = "Update not available"
    //       break
    //     case "UPDATE_ERROR":
    //       message = "An error occurred while attempting to update"
    //       break
    //     case "UPDATE_DOWNLOADED":
    //       message = "Update was successfully downloaded"
    //       break
    //   }
    //
    //   this.$notify.info({ title, message })
    // })
  }
}
</script>
