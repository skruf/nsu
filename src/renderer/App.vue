<i18n>
{
  "en": {
    "appWarning": "This is a preview of the software and therefore errors might occur."
  },
  "no": {
    "appWarning": "Dette er en forhåndsvisning av programvaren og feil kan derfor oppstå."
  }
}
</i18n>

<style lang="stylus">
#app
  flex 1
  min-height 100vh
  display flex
  &:before
    content ""
    height 4px
    width 100%
    background-image var(--accent-border)
    position absolute
    top 0
    left 0
    right 0
    z-index 1
    // border-top-left-radius 3px
    // border-top-right-radius 3px

.titlebar
  -webkit-user-select none
  -webkit-app-region drag
  width 100%
  height 20px
  position absolute
  left 0
  right 0
  top 0

.app-warning
  z-index 1
  height 25px
  padding 0
  border-radius 0
  margin-bottom -25px
  .el-alert__title
    font-size 12px
    font-weight 400
  .el-alert__closebtn
    top 6px
    font-size 14px

// #app
//   padding-top 25px

</style>

<template>
  <div>
    <!-- <el-alert
      class="app-warning"
      type="warning"
      :title="$t('appWarning')"
      :closable="false"
      center
    /> -->

    <el-container
      id="app"
      :class="{ 'sidebar-open': isSidebarOpen }"
    >
      <div class="titlebar" />

      <app-sidebar
        :open.sync="isSidebarOpen"
      />

      <app-onboarding-dialog
        :shown.sync="appOnboardingShowDialog"
      />

      <router-view />
    </el-container>
  </div>
</template>

<script>
import AppOnboardingDialog from "@/components/AppOnboardingDialog"
import AppSidebar from "@/components/AppSidebar"

export default {
  name: "Nsu",

  components: {
    AppOnboardingDialog,
    AppSidebar
  },

  data: () => ({
    isSidebarOpen: true,
    appOnboardingShowDialog: false // true
  }),

  watch: {
    isSidebarOpen() {
      localStorage.setItem("isSidebarOpen", this.isSidebarOpen)
    },
    appOnboardingShowDialog() {
      localStorage.setItem("appOnboardingHideDialog", this.appOnboardingShowDialog)
    }
  },

  created() {
    this.isSidebarOpen = localStorage.getItem("isSidebarOpen") === "true"
    // this.appOnboardingShowDialog = !localStorage.getItem("appOnboardingHideDialog") === "true"
    this.$ipc.send("APP_STARTED")
    this.$ipc.on("APP_NOTIFICATION", (event, message) => {
      this.$notify.info({ title: "Notification", message: message })
    })
  }
}
</script>
