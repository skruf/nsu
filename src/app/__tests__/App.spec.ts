import VueRouter from "vue-router"
import Element from "element-ui"
import createLocalVue from "~/utils/tests/createLocalVue"
import mount from "~/utils/tests/mount"
import AppOnboardingDialog from "~/components/AppOnboardingDialog"
import AppSidebar from "~/components/AppSidebar"
import App from "~/App"

const localVue = createLocalVue([ VueRouter, Element ])

const build = (deep = false, overrides = {}) => {
  return mount(App, localVue, overrides, deep)
}

Storage.prototype.setItem = jest.fn().mockResolvedValue(true)

describe("App", () => {
  afterEach(() => {
    jest.clearAllMocks()
  })

  it("should render the component", () => {
    const wrapper = build()
    expect(wrapper.html()).toMatchSnapshot()
  })

  it("should have sidebar and onboarding", () => {
    const wrapper = build()
    expect(wrapper.find(AppOnboardingDialog).exists()).toBe(true)
    expect(wrapper.find(AppSidebar).exists()).toBe(true)
  })

  it("should display sidebar and onboarding by default", () => {
    const wrapper = build()
    expect(wrapper.vm.$data.isSidebarOpen).toEqual(true)
    expect(wrapper.vm.$data.appOnboardingShowDialog).toEqual(true)
  })

  it("should store display state of sidebar and onboarding", () => {
    const wrapper = build()
    wrapper.setData({
      isSidebarOpen: false,
      appOnboardingShowDialog: false
    })
    expect(localStorage.setItem).toHaveBeenCalledWith(
      "isSidebarOpen", false
    )
    expect(localStorage.setItem).toHaveBeenCalledWith(
      "appOnboardingShowDialog", false
    )
  })
})
