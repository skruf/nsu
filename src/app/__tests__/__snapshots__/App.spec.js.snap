// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App should render the component 1`] = `
<div>
  <el-alert-stub title="appWarning" description="" type="warning" closetext="" center="true" effect="light" class="app-warning"></el-alert-stub>
  <el-container-stub id="app" class="sidebar-open">
    <div class="titlebar"></div>
    <app-sidebar-stub open="true"></app-sidebar-stub>
    <app-onboarding-dialog-stub shown="true"></app-onboarding-dialog-stub>
    <router-view-stub name="default"></router-view-stub>
  </el-container-stub>
</div>
`;
